File: Learnify.Service/Implementation/AnswerService.cs
using Learnify.Domain.DTO;
using Learnify.Domain.Entities;
using Learnify.Repository.Interfaces;
using Learnify.Service.Interfaces;
using Learnify.Service.Mappers;
using Microsoft.EntityFrameworkCore;

namespace Learnify.Service.Implementation
{
    public class AnswerService : IAnswerService
    {
        private readonly IAnswerRepository repository;

        public AnswerService(IAnswerRepository repository)
        {
            this.repository = repository;
        }

        public Task DeleteAnswer(Guid answerId)
        {
            var answer = repository.Get(answerId);
            repository.Delete(answer);

            return Task.CompletedTask;
        }

        public Answer GetAnswer(Guid answerId)
        {
            var answer = repository.Get(answerId);

            return answer;
        }

        public async Task<List<Answer>> GetAnswers(Guid questionId)
        {
            var answers = await repository.GetAnswersAsync(questionId).ToListAsync();

            return answers;
        }

        public Task InsertAnswer(AnswerRequest answerRequest)
        {
            var answer = AnswerMapper.CopyFromDto(answerRequest);

            repository.Insert(answer);

            return Task.CompletedTask;
        }

        public Task UpdateAnswer(UpdateAnswer updateAnswer)
        {
            var answer = AnswerMapper.CopyFromUpdateDto(updateAnswer);

            repository.Update(answer);

            return Task.CompletedTask;
        }
    }
}

File: Learnify.UI/src/services/AnswerService.js
import { axios_config } from '../axios/axios_config';

const endpoint = '/answers';

export const getAnswers = async (questionId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${questionId}`);

        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err}`);

        throw err;
    }
};

export const getAnswer = async (answerId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${answerId}/get`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err}`);

        throw err;
    }
};

export const deleteAnswer = async (answerId) => {
    try {
        const response = await axios_config.delete(`${endpoint}/${answerId}`);
        
        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err}`);

        throw err;
    }
};

export const insertAnswer = async (answerRequest) => {
    try {
        const response = await axios_config.post(`${endpoint}`, answerRequest);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err}`);

        throw err;
    }
};

export const updateAnswer = async (updateAnswer) => {
    try {
        const response = await axios_config.put(`${endpoint}`, updateAnswer);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err}`);

        throw err;
    }
};
